Sources/Sentry/SentryClient.m

```diff
@@ -68,6 +67,19 @@ - (_Nullable instancetype)initWithOptions:(SentryOptions *)options
     return self;
 }
 
+/** Internal constructor for testing */
+- (instancetype)initWithOptions:(SentryOptions *)options
+                   andTransport:(id<SentryTransport>)transport
+                 andFileManager:(SentryFileManager *)fileManager
+{
+    self = [self initWithOptions:options];
+
+    self.transport = transport;
+    self.fileManager = fileManager;
+
+    return self;
+}
+
```

Comment:
it might be possible if you move these fields declaration to the interface (the real one).
and then `SentryClient+TestInit.h ` also implements the real interface, so both would have access to the fields, just a hunch though.